core/src/main/java/org/fao/geonet/kernel/metadata/DefaultStatusActions.java

```diff
@@ -369,6 +369,25 @@ protected List<User> getUserToNotify(MetadataStatus status) {
                 return new ArrayList<>();
         }
 
+        // If status is DRAFT and previous status is SUBMITTED, which means either:
+        //   - a cancel working copy (from editor) --> should be notified the reviewer.
+        //   - rejection (from reviewer) --> should be notified the editor.
+        // and the notification level is recordUserAuthor or recordProfileReviewer,
+        // then adjust the notification level, depending on the user role
+        if ((status.getStatusValue().getId() == Integer.parseInt(StatusValue.Status.DRAFT)) &&
+            (!StringUtils.isEmpty(status.getPreviousState()) &&
+                (Integer.parseInt(status.getPreviousState()) == Integer.parseInt(StatusValue.Status.SUBMITTED))) &&
```

Comment:
Any reason the following would not work?
```suggestion
                (status.getPreviousState().equals(StatusValue.Status.SUBMITTED))  &&
```